src/v/cluster/partition_leaders_table.cc

```diff
@@ -88,29 +97,51 @@ void partition_leaders_table::update_partition_leader(
         auto [new_it, _] = _leaders.emplace(
           leader_key{
             model::topic_namespace(ntp.ns, ntp.tp.topic), ntp.tp.partition},
-          leader_meta{.current_leader = leader_id, .update_term = term});
+          leader_meta{
+            .current_leader = leader_id,
+            .update_term = term,
+            .partition_revision = revision_id});
         it = new_it;
     } else {
+        // Currently we have to check if revision id is valid since not all the
+        // code paths devlivers revision information
+        //
+        // TODO: always check revision after we will add revision to metadata
+        // dissemination requests
+        const bool revision_id_valid = revision_id >= model::revision_id{0};
+        // skip update for partition with previous revision
+        if (revision_id_valid && revision_id < it->second.partition_revision) {
+            return;
+        }
+
         // existing partition
-        if (it->second.update_term > term) {
-            // Do nothing if update term is older
+        if (
+          it->second.update_term > term
+          && (!revision_id_valid || it->second.partition_revision >= revision_id)) {
+            // Do nothing if update term is older and revision matches or
+            // revision_id is invalid
             return;
         }
+
         // if current leader has value, store it as a previous leader
         if (it->second.current_leader) {
             it->second.previous_leader = it->second.current_leader;
         }
         it->second.current_leader = leader_id;
         it->second.update_term = term;
+        if (revision_id_valid) {
+            it->second.partition_revision = revision_id;
+        }
     }
     vlog(
```

Comment:
are we also logging enough information before this to be able to helpful in debugging? for example in several cases above we return early so won't hit this log statement.